modules/core/src/test/java/org/apache/ignite/internal/client/thin/TimeoutTest.java

```diff
@@ -217,4 +217,120 @@ public void testClientTimeoutOnOperation() throws Exception {
             }
         }
     }
+
+    /**
+     * Test that connection timeout is independent of request timeout during connection establishment.
+     */
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testConnectionTimeoutIndependentOfRequest() throws Exception {
+        ServerSocket sock = new ServerSocket();
+        sock.bind(new InetSocketAddress("127.0.0.1", DFLT_PORT));
+
+        CountDownLatch connectionAccepted = new CountDownLatch(1);
+
+        IgniteInternalFuture<?> fut = GridTestUtils.runAsync(() -> {
+            try {
+                Socket accepted = sock.accept();
+                connectionAccepted.countDown();
+
+                Thread.sleep(2000);
+
+                U.closeQuiet(accepted);
+            }
+            catch (Exception e) {
+                throw new IgniteException("Accept thread failed: " + e.getMessage(), e);
+            }
+        });
+
+        long startTime = System.currentTimeMillis();
+
+        try {
+            ClientConfiguration cfg = new ClientConfiguration()
+                .setAddresses("127.0.0.1:" + DFLT_PORT)
+                .setConnectionTimeout(500)
+                .setRequestTimeout(10000);
+
+            GridTestUtils.assertThrowsWithCause(
+                () -> Ignition.startClient(cfg),
+                ClientConnectionException.class
+            );
+        }
+        finally {
+            U.closeQuiet(sock);
+        }
+
+        long elapsed = System.currentTimeMillis() - startTime;
+
+        assertTrue("Should fail around connection timeout (500ms), not request timeout",
```

Comment:
This checks are not stable and must be avoided:
1. System.currentTimeMillis() returns the wall-clock time, which is non-monotonic and susceptible to adjustments, such as NTP synchronization or manual system clock updates
2. Thread might be blocked more than 2s. 

Just set request timeout to Integer.MAX_VALUE. Also server handshake time must be overridden - see `getConfiguration` in this class.